lib/livebook/hubs/team_client.ex

```diff
@@ -357,7 +368,29 @@ defmodule Livebook.Hubs.TeamClient do
     |> dispatch_deployment_groups(user_connected)
   end
 
+  defp handle_event(:agent_connected, agent_connected, state) do
+    %{state | deployment_groups: [agent_connected.deployment_group]}
+    |> dispatch_secrets(agent_connected)
+    |> dispatch_file_systems(agent_connected)
```

Comment:
This brings the question: if the deployment group is updated, should it update agents live? I would say that yes, new deployments should use the latest secrets.

Reply:
Yes, I would say that we should always have the Agents up-to-date.